### ISSUE TITLE
Adding `--Discovery.DiscV5Enabled true` to a synced DB causes Nethermind to fail

### ISSUE THREAD

**Issue by yorickdowne** [NethermindEth/nethermind#7519]
**Description**

On a synced DB - tested on Ethereum mainnet and Holesky - add `--Discovery.DiscV5Enabled true`. Nethermind will then fail, logs see below.

A workaround is to remove or rename `nethermind_db/$NETWORK/discoveryNodes`. The directory will be recreated and Nethermind won't fail.

**Desktop (please complete the following information):**
Please provide the following information regarding your setup:
 - Operating System: Linux
 - Version: 1.28.0
 - Installation Method: Docker
  - Consensus Client: any

**Logs**
```
execution-1  | 30 Sep 20:23:45 | Detected HalfPath key scheme. 
execution-1  | 30 Sep 20:23:45 | Snap serving enabled, but PruningBoundary is less than 128. Setting to 128. 
execution-1  | 30 Sep 20:23:45 | Snap - State Ranges (Phase 1) is finished. 
execution-1  | 30 Sep 20:23:45 | Pivot block has been set based on data from db. Pivot block number: 1416579, hash: 0x9808de3e7470c51cb83f1524f5cadbaa5b8e5fd2ef2dbfb8ca27230d99196997 
execution-1  | 30 Sep 20:23:45 | Step InitializeNetwork        failed after 54ms System.ArgumentException: Input byte array must be no more than 8 bytes long. (Parameter 'bytes')
execution-1  |    at Lantern.Discv5.Rlp.RlpExtensions.ByteArrayToNumber[T](Byte[] bytes, Int32 targetLength, Func`3 converter)
execution-1  |    at Lantern.Discv5.Enr.EnrFactory.CreateFromDecoded(IReadOnlyList`1 items, IIdentityVerifier verifier)
execution-1  |    at System.Linq.Enumerable.SelectIListIterator`2.MoveNext()
execution-1  |    at System.Collections.Generic.List`1.AddRange(IEnumerable`1 collection)
execution-1  |    at Nethermind.Network.Discovery.DiscoveryV5App..ctor(SameKeyGenerator privateKeyProvider, IApiWithNetwork api, INetworkConfig networkConfig, IDiscoveryConfig discoveryConfig, SimpleFilePublicKeyDb discoveryDb, ILogManager logManager) in /src/Nethermind/Nethermind.Network.Discovery/Discv5/DiscoveryV5App.cs:line 65
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.InitDiscovery() in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 319
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.Initialize(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 184
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.Execute(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 85
execution-1  |    at Nethermind.Init.Steps.EthereumStepsManager.ExecuteStep(IStep step, StepInfo stepInfo, CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs:line 153
execution-1  |    at Nethermind.Init.Steps.EthereumStepsManager.InitializeAll(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs:line 95
execution-1  |    at Nethermind.Runner.Ethereum.EthereumRunner.Start(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Runner/Ethereum/EthereumRunner.cs:line 36
execution-1  |    at Nethermind.Runner.Program.<>c__DisplayClass8_0.<<Run>b__1>d.MoveNext() in /src/Nethermind/Nethermind.Runner/Program.cs:line 213
execution-1  | 30 Sep 20:23:45 | Error during ethereum runner start System.ArgumentException: Input byte array must be no more than 8 bytes long. (Parameter 'bytes')
execution-1  |    at Lantern.Discv5.Rlp.RlpExtensions.ByteArrayToNumber[T](Byte[] bytes, Int32 targetLength, Func`3 converter)
execution-1  |    at Lantern.Discv5.Enr.EnrFactory.CreateFromDecoded(IReadOnlyList`1 items, IIdentityVerifier verifier)
execution-1  |    at System.Linq.Enumerable.SelectIListIterator`2.MoveNext()
execution-1  |    at System.Collections.Generic.List`1.AddRange(IEnumerable`1 collection)
execution-1  |    at Nethermind.Network.Discovery.DiscoveryV5App..ctor(SameKeyGenerator privateKeyProvider, IApiWithNetwork api, INetworkConfig networkConfig, IDiscoveryConfig discoveryConfig, SimpleFilePublicKeyDb discoveryDb, ILogManager logManager) in /src/Nethermind/Nethermind.Network.Discovery/Discv5/DiscoveryV5App.cs:line 65
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.InitDiscovery() in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 319
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.Initialize(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 184
execution-1  |    at Nethermind.Init.Steps.InitializeNetwork.Execute(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs:line 85
execution-1  |    at Nethermind.Init.Steps.EthereumStepsManager.ExecuteStep(IStep step, StepInfo stepInfo, CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs:line 153
execution-1  |    at Nethermind.Init.Steps.EthereumStepsManager.InitializeAll(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs:line 95
execution-1  |    at Nethermind.Runner.Ethereum.EthereumRunner.Start(CancellationToken cancellationToken) in /src/Nethermind/Nethermind.Runner/Ethereum/EthereumRunner.cs:line 36
execution-1  |    at Nethermind.Runner.Program.<>c__DisplayClass8_0.<<Run>b__1>d.MoveNext() in /src/Nethermind/Nethermind.Runner/Program.cs:line 213
execution-1  | 30 Sep 20:23:45 | Closing, please wait until all functions are stopped properly... 
execution-1  | 30 Sep 20:23:45 | Stopping session monitor... 
```

**Comment by yorickdowne** [comment]
Also true for `--Discovery.DiscoveryVersion=V5` on `1.29.0`

**Comment by tanishqjasoria** [comment]
Fixed by #7511
